sale_quotation_number/models/sale_order.py

```diff
@@ -2,29 +2,27 @@
 # © 2010-2012 Andy Lu <andy.lu@elico-corp.com> (Elico Corp)
 # © 2013 Agile Business Group sagl (<http://www.agilebg.com>)
 # © 2017 valentin vinagre  <valentin.vinagre@qubiq.es> (QubiQ)
+# © 2020 Manuel Regidor  <manuel.regidor@sygel.es>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
 
-from odoo import models, api, fields
-
-
-class ResCompany(models.Model):
-    _inherit = 'res.company'
-
-    keep_name_so = fields.Boolean(
-        string="Use Same Enumeration",
-        help="If this is unchecked, quotations use a different sequence from "
-             "sale orders", default=True)
-
-
-class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
-
-    keep_name_so = fields.Boolean(related='company_id.keep_name_so')
+from odoo import models, api
 
 
 class SaleOrder(models.Model):
     _inherit = "sale.order"
 
+    @api.model
+    def create(self, vals):
+        company = vals.get('company_id', False)
+        if company:
+            company = self.env['res.company'].browse(company)
+        else:
+            company = self.env['res.company']._company_default_get('sale.order')
```

Comment:
@manuelregidor I think here you should add direct `company = self.env.user.company_id`

Reply:
That's what function _company_default_get() does, so I'd leave it the way it is now.

Reply:
@manuelregidor  Yes you are right 
But IMO `_company_default_get()` method called `self.env['res.users']._get_company()` method and `return self.env.user.company_id`.

That's why I am asking to put directly `self.env.user.company_id`

Reply:
If other module inherits the `_company_default_get `, IMO it's not a bad solution.